compiler/rustc_errors/src/emitter.rs

```diff
@@ -1410,6 +1406,42 @@ impl EmitterWriter {
         for annotated_file in annotated_files {
             // we can't annotate anything if the source is unavailable.
             if !sm.ensure_source_file_source_present(annotated_file.file.clone()) {
+                if !self.short_message {
+                    // We'll just print an unannotated message.
```

Comment:
Here is where we'd have to duplicate some of the printing logic for accurate gutter alignment. :-/

Ideally we'd dedup the logic, but it is fine as long as we have test coverage (which we do) and leave a comment in both places reminding us to modify logic in the other.